src/components/Radio/index.tsx

```diff
@@ -1,145 +1,125 @@
-  color: ${({ theme }) => theme.colors.danger.text};
-  padding: ${({ theme }) => `0 ${theme.space['0.5']}`};
+  ${StyledRadio}[aria-invalid="true"]:focus  + ${StyledIcon} {
+    background-color: ${({ theme }) => theme.colors.danger.background};
+    fill: ${({ theme }) => theme.colors.danger.text};
+  }
+
+  :hover[aria-disabled='false'] {
+    ${StyledRadio} + ${StyledIcon} {
+      background-color: ${({ theme }) => theme.colors.primary.background};
+      fill: ${({ theme }) => theme.colors.primary.text};
+    }
+
+    ${StyledRadio}[aria-invalid="true"]  + ${StyledIcon} {
+      background-color: ${({ theme }) => theme.colors.danger.background};
+      fill: ${({ theme }) => theme.colors.danger.text};
+    }
+  }
 `
 
 type RadioProps = {
   children: ReactNode
-  valid?: boolean
   error?: string | ReactNode
   name: string
   size?: number
   value: string | number
-  showError?: boolean
-} & XStyledProps &
-  InputHTMLAttributes<HTMLInputElement>
+} & InputHTMLAttributes<HTMLInputElement>
 
 const Radio = ({
-  checked,
+  checked = false,
   onChange,
   onFocus,
   onBlur,
   disabled = false,
   error,
   name,
-  valid,
   value,
-  size = 24,
+  size = 20,
   children,
-  showError = true,
-  ...props
-}: RadioProps): JSX.Element => {
-  const color = useMemo(() => {
-    if (disabled) return 'gray100'
-    if (valid === false || !!error) return 'warning'
-    if (valid === true) return 'success'
-    if (checked) return 'primary'
-
-    return 'gray300'
-  }, [disabled, valid, checked, error])
-
-  return (
-    <Box {...props}>
-      <StyledRadioContainer
-        as="label"
-        htmlFor={`${name}-${value}`}
-        disabled={disabled}
-      >
-        <IconContainer>
-          <StyledIcon
-            name={checked ? 'radiobox-marked' : 'radiobox-blank'}
-            color={color}
-            size={size}
-          />
-        </IconContainer>
-        <div>{children}</div>
-        <StyledRadio
-          type="radio"
-          aria-checked={checked ? 'true' : 'false'}
-          checked={checked}
-          id={`${name}-${value}`}
-          onChange={onChange}
-          onFocus={onFocus}
-          onBlur={onBlur}
-          value={value}
-          disabled={disabled}
-          name={name}
-        />
-      </StyledRadioContainer>
-      {showError ? (
-        <Expandable opened={!!error}>
-          <StyledError>{error}</StyledError>
-        </Expandable>
-      ) : null}
-    </Box>
-  )
-}
+  className,
+  autoFocus,
+}: RadioProps): JSX.Element => (
+  <StyledRadioContainer
+    as="label"
+    aria-disabled={disabled}
+    htmlFor={`${name}-${value}`}
+    className={className}
+  >
+    <StyledRadio
+      type="radio"
+      aria-invalid={!!error}
+      aria-checked={checked}
+      checked={checked}
+      id={`${name}-${value}`}
+      onChange={onChange}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      value={value}
+      disabled={disabled}
+      name={name}
+      autoFocus={autoFocus}
```

Comment:
Keyboard interaction is missing for a11y, please check this documentation: https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/radio/radio-1/radio-1.html

Reply:
`onKeyDown`? like that in the checkbox ?

Reply:
@Lawndlwd yes exactly, it should handle all those keys:

![Screenshot 2022-05-24 at 09 38 12](https://user-images.githubusercontent.com/15812968/169975690-250ba6ea-0e67-4e11-99de-3fbab7cc2ce7.png)

Reply:
Resolved, because we need a radio group to control radio to be able to handle keyDown's